src/basic/memory-util.c

```diff
@@ -39,3 +39,17 @@ bool memeqbyte(uint8_t byte, const void *data, size_t length) {
         /* Now we know first 16 bytes match, memcmp() with self.  */
         return memcmp(data, p + 16, length) == 0;
 }
+
+void *make_swapped(const void *data, size_t size) {
+        assert(data);
+        assert(size != 0);
+
+        void *p = malloc(size);
+        if (p) {
```

Comment:
we usually want that in success checks the "main" codepath is the one that is not indented, and the exceptional one is the one that is indented.

i.e. please change this to: `if (!p) return NULL;`

you get the idea